src/main/io/gps_ublox.c

```diff
@@ -823,9 +823,13 @@ STATIC_PROTOTHREAD(gpsConfigure)
     // Configure UBX binary messages
     gpsSetProtocolTimeout(GPS_SHORT_TIMEOUT);
 
-    // u-Blox 9
-    // M9N does not support some of the UBX 6/7/8 messages, so we have to configure it using special sequence
-    if (gpsState.hwVersion >= 190000) {
+    // M9N & M10 does not support some of the UBX 6/7/8 messages, so we have to configure it using special sequence
+    // ublox_6   hwVersion 00040007
+    // ublox_7   hwVersion 00070000
+    // ublox_M8  hwVersion 00080000
+    // ublox_M9  hwVersion 00190000
+    // ublox_M10 hwVersion 000A0000
+    if (gpsState.hwVersion > 80000) {
```

Comment:
Shouldn't this be a hex comparison `0x80000` (and it was just good fortune that the M9N check worked).?

Reply:
It appears that the value is actually hexadecimal and it's a mere chance that it worked before

Reply:
Actually it's even worse than that. There are no guarantees that the hw version is a number, it's a string and I think we should better be treating it as such.